keel-plugin/src/main/kotlin/com/netflix/spinnaker/keel/plugin/ResourceHandler.kt

```diff
@@ -171,6 +169,34 @@ abstract class ResourceHandler<S : ResourceSpec, R : Any>(
    * associated with [id], `false` otherwise.
    */
   open suspend fun actuationInProgress(id: ResourceId): Boolean = false
+
+  /**
+   * Used to register the [ResourceSpec] sub-type supported by this handler with Jackson so we can
+   * serialize and deserialize it. Do not override this or call it, or even look at it. You never
+   * saw this method, alright?
+   */
+  fun registerResourceKind(objectMappers: Iterable<ObjectMapper>) {
+    with(supportedKind) {
+      log.info("Registering ResourceSpec sub-type {}: {}", typeId, specClass.simpleName)
+      objectMappers.forEach { it.registerSubtypes(namedType) }
+    }
+  }
+
+  /**
+   * Convenient version of `registerResourceKind(Iterable<ObjectMapper>)` for tests, etc.
+   */
+  fun registerResourceKind(vararg objectMappers: ObjectMapper) {
+    registerResourceKind(objectMappers.toList())
+  }
+}
+
+data class SupportedKind<SPEC : ResourceSpec>(
+  val apiVersion: ApiVersion,
+  val kind: String,
+  val specClass: Class<SPEC>
+) {
+  val typeId = "$apiVersion/$kind"
+  val namedType = NamedType(specClass, typeId)
 }
```

Comment:
Data class to indicate what type of resource a handler supports. Handy properties to convert to Jackson-y things.